sandbox/libs/authn/src/main/java/org/opensearch/authn/internal/InternalSubject.java

```diff
@@ -64,9 +64,15 @@ public String toString() {
      * Logs the user in via authenticating the user against current Shiro realm
      */
     public void login(AuthenticationToken authenticationToken) {
-        org.apache.shiro.authc.AuthenticationToken authToken = AuthenticationTokenHandler.extractShiroAuthToken(authenticationToken);
+
+        org.apache.shiro.authc.AuthenticationToken authToken = null;
+        try {
+            authToken = AuthenticationTokenHandler.extractShiroAuthToken(authenticationToken);
+        } catch (BadCredentialsException e) {
+            throw new RuntimeException(e);
```

Comment:
This try/catch is no longer needed as handleBearerAuth will no longer throw BadCredentialsException and by extension this method too.

Reply:
After removing the the throw from the handleBearerAuth method, we still need most of these catches and throws where they are. We need to be able to specify the exception type thrown by AuthenticationTokenHandler so that we can catch them in testing without a generic Exception-based try/catch. Perhaps you had another thought for organizing things and where you would suggest to throw the BadCredentialsException but right now it still needs to be thrown from extractShiroToken.

Reply:
All that `handleBearerAuth` is doing is:

```
private static AuthenticationToken handleBearerAuth(final BearerAuthToken token) {

    String encodedJWT = token.getHeaderValue().substring("Bearer".length()).trim();
    return new BearerToken(encodedJWT);
}
```

No exception is through within there except for a possible NPE is token is null as @DarshitChanpura pointed out.

You can remove `throws BadCredentialsException` from `extractShiroToken`

Reply:
Thank you both for the time spent providing feedback. I know the throws can be removed from `extractShiroToken` but I do not know what is suggested for an alternative to the current testing format. 

My understanding is that we are standardizing the test cases to match each other and be able to use asserts in order to compare error messages against expected output--we want to assert that certain behavior is caught instead of just checking that the base behavior works as intended. I am not sure where to refactor the throw to so that we can still catch off the throw in the test cases. 

Example from the code: 

```
 try {
            AuthenticationToken extractedShiroToken = extractShiroAuthToken(bearerAuthToken);
        } catch (BadCredentialsException ex) {
            assertFalse(ex.getMessage().isEmpty());
            assertEquals("The token has expired", ex.getMessage());
        }
```

Unless we want to swap the BadCredentialsException type to a generic Exception then something needs to be modified before removing the throw from the extractShiroToken method. We (I) need to move the throw BadCredentialsException elsewhere. I do not know where a good place to add this throw would be. Alternatively, I can remove all the edge case tests. 

Please let me know what you think would be best and I can take care of it!

Reply:
The top 4 tests in `BearerAuthTests` look like they are testing the JWT verification logic and can be refactored to use one of the methods in `JWTVerifier` to test JWT Verification code. The methods in there throw the `BadCredentialsException`

```
- testExpiredValidJwt
- testEarlyValidJwt
- testValidJwt
- testMismatchedSignatureJwt
```

Reply:
Duh, sorry for not doing this before. For some reason it just did not dawn on me to switch things in that manner. Thank you!